lib/HLSL/DxilValidation.cpp

```diff
@@ -2134,17 +2134,38 @@ static void ValidateBarrierFlagArg(ValidationContext &ValCtx, CallInst *CI,
   }
 }
 
+std::string GetLaunchTypeStr(DXIL::NodeLaunchType LT) {
+  switch (LT) {
+	case DXIL::NodeLaunchType::Broadcasting:
+		return "Broadcasting";
+  case DXIL::NodeLaunchType::Coalescing:
+		return "Coalescing";
+  case DXIL::NodeLaunchType::Thread:
+    return "Thread";
+  case DXIL::NodeLaunchType::Invalid:
+  case DXIL::NodeLaunchType::LastEntry:
+    return "Invalid";
+
+  default:
+    return "Invalid";
```

Comment:
Since you're covering all of the cases, it'd be better to skip the `default` label so that `-Wswitch` can warn if any enum values are missed (ie, if they're added). You'll need to add an `llvm_unreachable` call outside of the switch to quiet warnings about not all control paths returning a value on some compilers.